### PR TITLE
feat(APIv2): when assigning a system, make sure it has a tailoring

### PR DESCRIPTION
**Pull request by skateman**
## Secure Coding Practices Checklist GitHub Link
- https://github.com/RedHatInsights/secure-coding-checklist

## Secure Coding Checklist
- [ ] Input Validation
- [ ] Output Encoding
- [ ] Authentication and Password Management
- [ ] Session Management
- [ ] Access Control
- [ ] Cryptographic Practices
- [ ] Error Handling and Logging
- [ ] Data Protection
- [ ] Communication Security
- [ ] System Configuration
- [ ] Database Security
- [ ] File Management
- [ ] Memory Management
- [x] General Coding Practices

### CODE REVIEW

File: app/models/v2/policy_system.rb [hunk 1]
```diff
@@ -14,6 +14,17 @@ class PolicySystem < ApplicationRecord
     validates :system_id, presence: true, uniqueness: { scope: :policy_id }
     validate :system_supported?, on: :create
 
+    after_create do
+      variant = policy.profile.variant_for_minor(system.os_minor_version)
+
+      Tailoring.find_or_create_by!(
+        profile_id: variant.id, policy_id: policy_id, os_minor_version: system.os_minor_version
```
Comment (skateman) [inline app/models/v2/policy_system.rb:21]: Possible bug if a newer version of a profile comes out for the same `ref_id` and `is_minor_version`, so this has to be somehow composed into the creation.

**Comment by skateman** [comment]
@romanblanco resolved, ready for review...added a test for the edge-case described above

File: app/models/v2/profile.rb [hunk 2]
```diff
@@ -18,5 +18,14 @@ class Profile < ApplicationRecord
     has_many :profile_rules, class_name: 'V2::ProfileRule', dependent: :destroy
     has_many :rules, through: :profile_rules, class_name: 'V2::Rule'
     has_many :os_minor_versions, class_name: 'V2::ProfileOsMinorVersion', dependent: :destroy
+
+    def variant_for_minor(version)
+      self.class.joins(:security_guide, :os_minor_versions)
+          .order(self.class.version_to_array(V2::SecurityGuide.arel_table.alias('security_guide')[:version]).desc)
+          .find_by!(
+            ref_id: ref_id, security_guide: { os_major_version: security_guide.os_major_version },
```
Comment (romanblanco) [inline app/models/v2/profile.rb:26]: Please, indent the second parameter on a new line

**Review by romanblanco — APPROVED** [review]
LGTM :+1:
